### PR TITLE
fix: add error message in the responses

### LINKED ISSUE #721: Error message for entering Email id  which was not registered a user under "Sends the user a new verification email"
**Describe the bug**

When we enter mail id which is not registered any user, an appropriate error message should be shown and listed under the Responses.

**To Reproduce**

1. Go to 'user-Sends the user a new verification email'
2. Provide the email id
3. {
    "email": "soyiv65617@corsj.net"
    }
4. Click on 'Execute'.
5. Scroll down to 'Responses'.
6.  
Screenshot (280)

**Expected behavior**
That error message should be listed in the responses.

**Desktop (please complete the following information):**
 - OS: windows10
 - Browser chrome

### PR DESCRIPTION
**Pull request by ashwinijha6**
### Description
Added a valid error message in the responses for **user-Sends the user a new verification email** if the user is not registered.

Fixes #721 

### Type of Change:
- Code
- Quality Assurance

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
changed

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by ashwinijha6** [comment]
@SanketDG  & @PrashanthPuneriya needed a review on this let me know the changes.

**Comment by keshakaneria** [comment]
Seems good!

**Comment by ashwinijha6** [comment]
> Seems good!

Thanx for the review @keshakaneria I had a certain query
Should this message USER_IS_NOT_REGISTERED_IN_THE_SYSTEM be used as it is also used in the later part of the code ?

**Comment by ashwinijha6** [comment]
> > Seems good!
> 
> Thanx for the review @keshakaneria I had a certain query
> Should this message USER_IS_NOT_REGISTERED_IN_THE_SYSTEM be used as it is also used in the later part of the code ?
> 
> 

USER_IS_NOT_REGISTERED_IN_THE_SYSTEM as in messages.py is **You are not registered in the system**

**Review by SanketDG — DISMISSED** [review]
LGTM!

**Comment by keshakaneria** [comment]
> > Seems good!
> 
> Thanx for the review @keshakaneria I had a certain query
> Should this message USER_IS_NOT_REGISTERED_IN_THE_SYSTEM be used as it is also used in the later part of the code ?
> 
> 

It will be fine the way it is right now

**Comment by ashwinijha6** [comment]
> > > Seems good!
> > 
> > 
> > Thanx for the review @keshakaneria I had a certain query
> > Should this message USER_IS_NOT_REGISTERED_IN_THE_SYSTEM be used as it is also used in the later part of the code ?
> > 
> 
> It will be fine the way it is right now

ok @keshakaneria thanx

**Comment by ashwinijha6** [comment]
@keshakaneria could u pls put the required labels

**Review by keshakaneria — DISMISSED** [review]

**Comment by ashwinijha6** [comment]
@keshakaneria & @isabelcosta does this require any more changes?

**Comment by keshakaneria** [comment]
@isabelcosta Can you let us know whom should we mention here to get this PR tested as you've added in the tags?

**Comment by rpattath** [comment]
QA team will test this as soon as one of can get to it. @ashwinijha6 since this PR has 2 approvals it should be good to be qualitfied for OSH. You can move on to other issues.

**Comment by ashwinijha6** [comment]
> QA team will test this as soon as one of can get to it. @ashwinijha6 since this PR has 2 approvals it should be good to be qualitfied for OSH. You can move on to other issues.

Thanx @rpattath for the response

File: app/messages.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
     "message": "Mentorship relation does not" " exist."
 }
 USER_NOT_FOUND = {"message": "User not found."}
+USER_NOT_REGISTERED = {"message": "Please register as a user first to get a verification email."}
```
Comment (devkapilbansal) [inline app/messages.py]: The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses (positive and negative tests) were tested as below:

   * _Test1 Description_  
     _Screenshot/gif_:   Attached below
     _Expected Result_:  Message should be "You are not registered in the system"
     _Actual Result_: Message is "Please register as a user first to get a verification email."

4. Additional Comments: This PR needs changes.

Before changes : - 
before_testing

Actual Results:
tested

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
@ashwinijha6 please made the required changes

**Comment by devkapilbansal** [comment]
@isabelcosta this pr needs changes

**Comment by ashwinijha6** [comment]
> @ashwinijha6 please made the required changes

@devkapilbansal I have written this message purposely "Please register as a user first to get a verification email." It was also reviewed by @keshakaneria & @SanketDG  
I thought this would be a more appropriate message than  "You are not registered in the system"
What do u think @devkapilbansal & @isabelcosta ?

**Comment by devkapilbansal** [comment]
Actually, we need both messages to be same, as mentioned in the issue itself. Pinging @SanketDG for his review

**Comment by ashwinijha6** [comment]
> Actually, we need both messages to be same, as mentioned in the issue itself. Pinging @SanketDG for his review

Ok @devkapilbansal do u mean that the response body should also have the message, That's the point right?
Also which of the 2 messages should I go with** Please register** or **You are not registered** ?

**Comment by ashwinijha6** [comment]
@devkapilbansal should both the sentences be **You are not registered in the system** or **Please register as a user first to get a verification email.**?

**Comment by rpattath** [comment]
> @devkapilbansal should both the sentences be **You are not registered in the system** or **Please register as a user first to get a verification email.**?

Yes both should be the same whichever you choose to use.

**Comment by ashwinijha6** [comment]
> > @devkapilbansal should both the sentences be **You are not registered in the system** or **Please register as a user first to get a verification email.**?
> 
> Yes both should be the same whichever you choose to use.

ok I guess then I should go with **You are not registered in the system" as mentioned by @devkapilbansal 👍

**Comment by ashwinijha6** [comment]
@devkapilbansal should there be space between in & the in the sentence
reg

**Comment by devkapilbansal** [comment]
Yes yes, there should be.

**Comment by ashwinijha6** [comment]
@devkapilbansal I have made the required changes could u pls review it.

**Review by keshakaneria — DISMISSED** [review]
Looks good now 😄

File: app/messages.py (hunk 1 above)
Comment (devkapilbansal) [inline app/messages.py]: Is this message still required??

File: app/messages.py [hunk 2]
```diff
@@ -147,7 +148,7 @@
     "message": "A user with that " "email already exists."
 }
 USER_IS_NOT_REGISTERED_IN_THE_SYSTEM = {
```
Comment (devkapilbansal) [inline app/messages.py:149]: This name is a bit longer we can use `USER_NOT_REGISTERED` only here as you mentioned above. Remove that message and rename it. Also, it should be `"You are not registered in the system"`

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Thanks @ashwinijha6 look my request below

File: app/messages.py (hunk 1 above)
Comment (ashwinijha6) [reply app/messages.py]: Is this message still present I removed it?

File: app/messages.py (hunk 2 above)
Comment (ashwinijha6) [reply app/messages.py:149]: Ok I will change it

File: app/messages.py (hunk 1 above)
Comment (ashwinijha6) [reply app/messages.py]: ya its there

**Comment by ashwinijha6** [comment]
@devkapilbansal I'm confused that I should remove line 25 & change line 150 is this the changes u want me to do?
But I change line 150 it is #Relation Constraints & that would hamper the message where ever it is previously initialized as USER_NOT_REGISTERED_IN_THE_SYSTEM.

**Comment by devkapilbansal** [comment]
> @devkapilbansal I'm confused that I should remove line 25 & change line 150 is this the changes u want me to do?
> But I change line 150 it is #Relation Constraints & that would hamper the message where ever it is previously initialized as USER_NOT_REGISTERED_IN_THE_SYSTEM.

Ohh, Yes you are right then. Just removed that extra message

File: app/messages.py (hunk 2 above)
Comment (ashwinijha6) [reply app/messages.py:149]: > > @devkapilbansal I'm confused that I should remove line 25 & change line 150 is this the changes u want me to do?
> > But I change line 150 it is #Relation Constraints & that would hamper the message where ever it is previously initialized as USER_NOT_REGISTERED_IN_THE_SYSTEM.
> 
> Ohh, Yes you are right then. Just removed that extra message

ok

**Comment by ashwinijha6** [comment]
@devkapilbansal I have made the changes

File: app/messages.py [hunk 3]
```diff
@@ -147,7 +147,7 @@
...
-    "message": "You are not registered in" "the system."
+    "message": "You are not registered in" " the system."
```
Comment (devkapilbansal) [inline app/messages.py]: Please change message `"You are not registered in" " the system"` to `"You are not registered in the system."`. We don't need to concatenate two strings rather can use a single string for this.

**Review by devkapilbansal — CHANGES_REQUESTED** [review]

File: app/messages.py (hunk 3 above)
Comment (ashwinijha6) [reply app/messages.py]: ok

**Comment by ashwinijha6** [comment]
Done @devkapilbansal

**Review by devkapilbansal — APPROVED** [review]
Thanks @ashwinijha6 for making the necessary changes.
LGTM :+1:

The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses (positive and negative tests) were tested as below:

   * _Test1 Description_  
     _Screenshot/gif_:   Attached below
     _Expected Result_:  Message should be "You are not registered in the system"
     _Actual Result_: Message is same as expected message.

4. Additional Comments: This PR is ready to merge

Before changes : - 
before_testing

Actual Results:
tested

**Comment by devkapilbansal** [comment]
@rpattath please update its label too

**Comment by ashwinijha6** [comment]
> Thanks @ashwinijha6 for making the necessary changes.
> LGTM 👍
> 
> The changes made in this PR were tested locally. Following are the results:
> 
> 1. Code review - Done
> 2. All possible responses (positive and negative tests) were tested as below:
>    
>    * _Test1 Description_
>      _Screenshot/gif_:   Attached below
>      _Expected Result_:  Message should be "You are not registered in the system"
>      _Actual Result_: Message is same as expected message.
> 3. Additional Comments: This PR is ready to merge
> 
> Before changes : -
> before_testing
> 
> Actual Results:
> tested

Grt @devkapilbansal 👍

**Review by keshakaneria — APPROVED** [review]
Thanks @ashwinijha6 Looks good now 👍

**Comment by ashwinijha6** [comment]
> Thanks @ashwinijha6 Looks good now 👍

👍 @keshakaneria

**Comment by rpattath** [comment]
Thank you @ashwinijha6 and @devkapilbansal

**Review by isabelcosta — APPROVED** [review]
Great work @ashwinijha6 🎉 
Thank you so much for this contribution :)

**Comment by ashwinijha6** [comment]
> Great work @ashwinijha6 🎉
> Thank you so much for this contribution :)

Thank you for appreciating my work @isabelcosta ❤️

**Pull request merged**